File: java/SelectionSort.java
import java.util.Random;

public class SelectionSort {

    public static void sort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIdx]) {
                    minIdx = j;
                }
            }
            if (minIdx != i) {
                int tmp = arr[i];
                arr[i] = arr[minIdx];
                arr[minIdx] = tmp;
            }
        }
    }

    public static void main(String[] args) {
        int n = Integer.parseInt(args[0]);
        Random rnd = new Random();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = rnd.nextInt();
        }

        long start = System.nanoTime();
        sort(arr);
        long end = System.nanoTime();

        for (int i = 1; i < n; i++) {
            if (arr[i - 1] > arr[i]) {
                System.err.println("ERRO: array nao ordenado corretamente");
                System.exit(1);
            }
        }

        double ms = (end - start) / 1_000_000.0;
        System.out.println(ms);
    }
}

File: python/selection_sort.py
import random
import sys
import time


def selection_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        min_idx = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        if min_idx != i:
            arr[i], arr[min_idx] = arr[min_idx], arr[i]


def main():
    n = int(sys.argv[1])
    arr = [random.randint(-1_000_000_000, 1_000_000_000) for _ in range(n)]

    start = time.perf_counter()
    selection_sort(arr)
    end = time.perf_counter()

    for i in range(1, n):
        if arr[i - 1] > arr[i]:
            print("ERRO: array nao ordenado corretamente", file=sys.stderr)
            sys.exit(1)

    ms = (end - start) * 1000.0
    print(ms)


if __name__ == "__main__":
    main()
